feat: Assistant API에서 시나리오 아이디와 유저 아이디를 받을 수 있도록 엔드포인트를 설계

jayn2u opened the pull request:
- 시나리오와 유저아이디를 클라이언트에서 받는 것으로 이해했습니다.
- 그래서 클라이언트에서 해당 인자를 받을 수 있도록 설계를 했는데, 한 번 검토해주시면 감사하겠습니다.
- 그리고 이미 배포된 점을 고려해서 기존의 엔드포인트에 영향을 미치지 않도록 `v2`로 설계했는데, 엔드포인트를 이렇게 설계하는지 검토해주시면 감사하겠습니다.

jayn2u committed:
feat: Assistant API 개선 및 응답 모델 수정

- AssistantController에서 API 경로 및 응답 형식 변경
- AssistantResponse 모델의 필드 이름 변경
- AssistantService 인터페이스에 시나리오 ID 및 사용자 ID를 포함한 메서드 추가
- LlmProxyService에서 응답 처리 로직 개선 및 시뮬레이션 서비스 통합

jayn2u committed:
refactor: 불필요한 import 정리 및 코드 스타일 개선
- 사용되지 않는 import 제거
- import 순서 정렬 및 정리

kwonyj1022 commented on src/main/java/com/newworld/saegil/llm/controller/AssistantController.java:
```diff
@@ -107,19 +97,27 @@ public ResponseEntity<Resource> getAssistantAudioResponseFromUploadedAudioFile(
     public ResponseEntity<AssistantResponse> getAssistantTextResponseFromUploadedAudioFile(
             @RequestPart("file") final MultipartFile multipartFile,
-            @RequestParam(value = "thread_id", required = false) final String threadId
+            @RequestParam(value = "thread_id", required = false) final String threadId,
```
얘는 `@Parameter` 어노테이션이 없네요

kwonyj1022 commented on src/main/java/com/newworld/saegil/llm/model/AssistantResponse.java:
```diff
@@ -14,11 +14,11 @@
 @ToString
 public class AssistantResponse {
```
달 데가 없어서 여기다 달아요. 저번에 급해서 리뷰 없이 머지하느라 못봤는데 model 말고 domain이라는 패키지명을 사용하고 있습니다.

kwonyj1022 replied:
이거 말고 이 패키지에 `TextToSpeechRequest`도 있는데 얘는 controller 레이어에서 사용하는, 클라이언트 요청을 받는 dto라 controller 패키지에 있어야 할 것 같습니다. 타입도 record 타입을 사용해주세요. 다른 쪽도 다 dto는 record를 사용하고 있어서요.

kwonyj1022 commented on src/main/java/com/newworld/saegil/llm/model/AssistantResponse.java:
```diff
@@ -14,11 +14,11 @@
 @ToString
 public class AssistantResponse {
```
그리고 따지고보면 dto라 record 타입으로 쓰면 좋을 것 같습니다. 또, 이게 fast api 서버의 response인지라 이게 과연 domain인가 하는 의문도 드네요. 서비스에서 리턴하는 dto에 가까운 것 같아서요. 만약 동의하신다면 패키지 위치도 service 패키지로 옮겨주시면 좋을 것 같습니다.

kwonyj1022 replied:
이거 record 타입으로 바꾸는거 적용이 안되어있어요!

jayn2u committed:
refactor: 불필요한 클래스 제거 및 패키지 구조 변경
- TextToSpeechRequest 클래스를 record로 변경하고 위치 변경
- AssistantResponse 클래스의 패키지를 model에서 service로 이동
- TextToSpeechRequest 관련 import 및 참조 수정
- 컨트롤러 메서드에 thread_id 및 scenario_id 파라미터 처리 수정

jayn2u committed:
refactor: ProxyProperties 코드 정렬 수정
- 필드 들여쓰기 정리로 코드 가독성 개선

kwonyj1022 reviewed (approved):
코멘트 답글 하나 남겨놨는데 먼저 확인하시면 적용해주시고 지금 불가능하시면 머지 전에 제가 바꿔놓을게요

kwonyj1022 committed:
refactor: AssistantResponse dto를 record 타입으로 변경
